climada/hazard/centroids/test/test_vec_ras.py

```diff
@@ -472,69 +300,81 @@ def test_get_closest_point(self):
-        self.assertAlmostEqual(centr_bis.meta['transform'].a, 0.009000000000000341)
-        self.assertAlmostEqual(centr_bis.meta['transform'].b, 0.0)
-        self.assertAlmostEqual(centr_bis.meta['transform'].f, 10.42822096697894)
-        self.assertAlmostEqual(centr_bis.meta['transform'].d, 0.0)
-        self.assertAlmostEqual(centr_bis.meta['transform'].e, -0.009000000000000341)
-        self.assertEqual(centr_bis.meta['height'], 60)
-        self.assertEqual(centr_bis.meta['width'], 50)
+    def test_from_meta(self):
+        """Test from_meta"""
+        meta_ref = {
+            'width': 10,
+            'height': 8,
+            'transform': rasterio.Affine(
+                0.6, 0, -0.1,
+                0, -0.6, 0.3,
+            ),
+            'crs': DEF_CRS,
+        }
+
+        lon_ref = np.array([0.2, 0.8, 1.4, 2.0, 2.6, 3.2, 3.8, 4.4, 5.0, 5.6])
+        lat_ref = np.array([0.0, -0.6, -1.2, -1.8, -2.4, -3.0, -3.6, -4.2])
+        lon_ref, lat_ref = [ar.ravel() for ar in np.meshgrid(lon_ref, lat_ref)]
+
+        centr = Centroids.from_meta(meta_ref)
+        meta = centr.get_meta()
+        self.assertTrue(u_coord.equal_crs(meta_ref["crs"], meta["crs"]))
+        self.assertEqual(meta_ref["width"], meta["width"])
+        self.assertEqual(meta_ref["height"], meta["height"])
+        np.testing.assert_allclose(meta_ref["transform"], meta["transform"])
+
+        centr = Centroids.from_meta(
+            Centroids(lat=lat_ref, lon=lon_ref).get_meta()
+        )
+        np.testing.assert_allclose(lat_ref, centr.lat)
+        np.testing.assert_allclose(lon_ref, centr.lon)
+
+        # `get_meta` enforces same resolution in x and y, and y-coordinates are decreasing.
+        # For other cases, `from_meta` needs to be checked manually.
+        meta_ref = {
+            'width': 4,
+            'height': 5,
+            'transform': rasterio.Affine(
+                0.5, 0, 0.2,
+                0, 0.6, -0.7,
+            ),
+            'crs': DEF_CRS,
+        }
+        lon_ref = np.array([0.45, 0.95, 1.45, 1.95])
+        lat_ref = np.array([-0.4, 0.2, 0.8, 1.4, 2.0])
+        lon_ref, lat_ref = [ar.ravel() for ar in np.meshgrid(lon_ref, lat_ref)]
+
+        centr = Centroids.from_meta(meta_ref)
+        np.testing.assert_allclose(lat_ref, centr.lat)
+        np.testing.assert_allclose(lon_ref, centr.lon)
 
     def test_equal_pass(self):
         """Test equal"""
         centr_ras = Centroids.from_raster_file(HAZ_DEMO_FL, window=Window(0, 0, 50, 60))
         centr_bis = Centroids.from_raster_file(HAZ_DEMO_FL, window=Window(51, 61, 10, 10))
-        self.assertFalse(centr_ras.equal(centr_bis))
-        self.assertFalse(centr_bis.equal(centr_ras))
-        self.assertTrue(centr_ras.equal(centr_ras))
-        self.assertTrue(centr_bis.equal(centr_bis))
+        self.assertFalse(centr_ras == centr_bis)
+        self.assertTrue(centr_ras == centr_ras)
+        self.assertTrue(centr_bis == centr_bis)
```

Comment:
Are you absolutely sure that this is checking the same thing, namely the `__eq__` dunder method of `Centroids`?

Reply:
Yes, I am. This is the code from `unittest.case` (with comments from me):

```python
    def _getAssertEqualityFunc(self, first, second):
        """Get a detailed comparison function for the types of the two args.

        Returns: A callable accepting (first, second, msg=None) that will
        raise a failure exception if first != second with a useful human
        readable error message for those types.
        """
        if type(first) is type(second):   # True, 1)
            asserter = self._type_equality_funcs.get(type(first))
            if asserter is not None:  # False, 2)
                if isinstance(asserter, str):
                    asserter = getattr(self, asserter)
                return asserter

        return self._baseAssertEqual

    def _baseAssertEqual(self, first, second, msg=None):
        """The default assertEqual implementation, not type specific."""
        if not first == second:
            standardMsg = '%s != %s' % _common_shorten_repr(first, second)
            msg = self._formatMessage(msg, standardMsg)
            raise self.failureException(msg)

    def assertEqual(self, first, second, msg=None):
        """Fail if the two objects are unequal as determined by the '=='
           operator.
        """
        assertion_func = self._getAssertEqualityFunc(first, second)
        assertion_func(first, second, msg=msg)
```

1) in our case both are `Centroids`
2) in our case, `Centroids` has never been registered with `TestCase.addTypeEqualityFunc()`, so we don't get a special Error message in case the test fails. See [https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual]

Reply:
I agree that it currently checks the same, but I'm not fully convinced that there is an advantage of using `assertEqual`. Instead, there is a very small risk that this implementation might silently change in the future - which is fine (because the risk is very small) if we at least have some kind of benefit for now. But if there is no benefit at all, then it's risky overall...